### ISSUE TITLE
Value Error at inference

### ISSUE THREAD

**Issue by ManuBN786** [openmedlab/MedLSAM#9]
![Screenshot from 2023-11-20 11-24-14](https://github.com/openmedlab/MedLSAM/assets/16796765/b331364d-b8f4-440a-a7b8-c7c939fdaeab)

How to resolve this error please?

Also I want to know if this model can be used to segment images outside the 22 classes mentioned in the paper.

I'm planning to use it for lung nodules segmentation.

**Comment by LWHYC** [comment]
> ![Screenshot from 2023-11-20 11-24-14](https://user-images.githubusercontent.com/16796765/284121263-b331364d-b8f4-440a-a7b8-c7c939fdaeab.png)
> 
> How to resolve this error please?
> 
> Also I want to know if this model can be used to segment images outside the 22 classes mentioned in the paper.
> 
> I'm planning to use it for lung nodules segmentation.

Hi, 
It seems like there is no fg pixel in your label. I suggest you check your support label data to make sure it contains your interested fg classes.
MedLSAM could be used to segment any anatomical structure and organs, but not the lesions and diseases. So it could not segment lung nodules.

**Comment by ManuBN786** [comment]
Thanks for the reply.

Can you please provide some training scripts and methodologies so one can train this model for nodule detection?

**Comment by LWHYC** [comment]
> Author

Hi,
Sorry for the late reply. We have added the training code here: https://github.com/openmedlab/MedLSAM/tree/main/train
